### DISCUSSION TITLE
can not find context inside required()

### DISCUSSION THREAD

**Discussion by Hegabovic** [filamentphp/filament#8154] [Help]
in filament version 2

i want a specific field to be required, so the red dot is shown 
but i want it to be hidden inside the view record context 

```
Forms\Components\TextInput::make('title')
                        ->label(__('Title'))
                        ->required()
```

i couldn't detect the context inside the required method , could you provide a code snippet for that

**Comment by leandrocfe** [comment]
in v3 there is an option
```php
->markAsRequired(false) // Removes the asterisk
```
in v2 you can use

```php
->required(fn (string $context): bool => $context !== 'view')
```

**Reply by Hegabovic** [reply]
i'm using filament version 2 , and it still being shown in the view 
even after i added this line 
```
->required(fn (string $context): bool => $context !== 'view')
```
![Screenshot from 2023-08-29 13-17-09](https://github.com/filamentphp/filament/assets/56083879/58e99815-6bb2-43e0-86f6-f8539cf80241)

**Reply by leandrocfe** [reply]
Did you add the view using this section?
https://filamentphp.com/docs/2.x/admin/resources/viewing-records

**Reply by Hegabovic** [reply]
yes sure i did it

**Reply by leandrocfe** [reply]
what about this?
```php
->required(fn (string $context): bool => Str::of($context)->contains(['create', 'edit']))
```

**Reply by Hegabovic** [reply]
the server required a restart in order to be able to view the change

**Reply by Hegabovic** [reply]
its all about restarting the livewire npm run dev
